gnome-initial-setup/pages/endless-eula/gis-endless-eula-viewer.c

```diff
@@ -0,0 +1,344 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright (C) 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
+ * 02111-1307, USA.
+ *
+ * Written by:
+ *     Damián Nohales <damian@endlessm.com>
+ */
+
+#include "config.h"
+#include "gis-endless-eula-viewer.h"
+
+#include <evince-document.h>
+#include <evince-view.h>
+#include <glib/gi18n.h>
+#include <locale.h>
+#include <webkit2/webkit2.h>
+
+typedef struct {
+  GtkWidget *evince_view;
+} GisEndlessEulaViewerPrivate;
+
+G_DEFINE_TYPE_WITH_PRIVATE (GisEndlessEulaViewer, gis_endless_eula_viewer, GTK_TYPE_SCROLLED_WINDOW)
```

Comment:
If you use `G_DECLARE_FINAL_TYPE` in the header then you can drop `_WITH_PRIVATE` and change the struct above to:

```c
struct _GisEndlessEulaViewer {
  GtkScrolledWindow parent;

  GtkWidget *evince_view;
};
```

and drop all references to the `Private` structure